IshiHisashi/GlucoFit
# feat: JDEV-83 - Creat screens for insight articles

## Description of this change
- create Insights screen and its related screens and integrate related APIs
- add a toast after creating a glucose log and integrate related API
- fix lazy data loading logic on Logs screen
- update the header on Home screen to show the user's name and today's date
- replace the line chart with a bar chart on Home screen
- the favourite buttons on Insights screens are NOT working yet. 
- the favourite tab on Insights screen is NOT working yet. 

## Steps to test
1. `cd frontend`, `npm install`, and run the app on an emulator
2. If necessary, `cd backend` and run the local server
3. You can visit Insights page to see the articles. 
4. You can create a glucose log from the sub menu, and you'll see a toast popping up after creating the log. You can tap on the toast to open the article. 

<img src="https://github.com/user-attachments/assets/655b7b99-37cf-454e-aae2-16170cecb148" width="150px" >
<img src="https://github.com/user-attachments/assets/ad166d4e-808a-4f31-b2d0-eab3a8a948f7" width="150px" >
<img src="https://github.com/user-attachments/assets/0b1d85e4-20d1-4f5d-b482-15ab4a83e38e" width="150px" >


## Ticket link
https://team-jupiter-capstone.atlassian.net/browse/JDEV-83
